iam-proxy-italia-project/backends/ciesaml2.py

```diff
@@ -156,8 +170,20 @@ def _metadata_endpoint(self, context):
         logger.debug("Sending metadata response")
         conf = self.sp.config
 
+        xmldoc = None
+
+        if self.xmldoc is not None:
+            logger.info(f"[INFO] xmldoc is not None: {self.xmldoc}")
+            xmldoc = self.xmldoc
+        else:
+            logger.info(f"[INFO] xmldoc is None: {self.xmldoc}")
+            xmldoc = self.create_metadata(conf)
+
+        logger.info(f"[INFO] xmldoc: {xmldoc}")
```

Comment:
```suggestion
        
        if not self.xmldoc:
            logger.debug(f"inizializing metadata xmldoc")
            self.xmldoc = self.create_metadata(conf)

        logger.debug(f"Created metadata xmldoc: {xmldoc}")
```

Anyway, it would be better to simply create self.xmldoc within the constructor of the class.

please do not print debug information in ordinary informations, log size matters in production environments.

please do not put  `[INFO]` or `[DEBUG]` as a string constant ... It MUST be loaded by the logger configuration.